packages/client/README.md

```diff
@@ -18,51 +18,133 @@ This library allows you to easily interact with the [Streamr Network](https://st
 
 [Installation](#installation) · [Usage](#usage) · [API Docs](#API-docs) · [Client options](#client-options) · [Authentication](#authentication-options) · [Managing subscriptions](#managing-subscriptions) · [Stream API](#stream-api) · [Subscription options](#subscription-options) · [Storage](#storage) ·[Data Unions](#data-unions) · [Utility functions](#utility-functions) · [Events](#events) · [Stream permissions](#stream-permissions) · [Stream Partitioning](#stream-partitioning) · [Logging](#logging) · [NPM Publishing](#publishing-latest)
 
+## Get Started
+Here are some usage examples. More examples can be found [here](https://github.com/streamr-dev/examples).
+
+> In Streamr, Ethereum accounts are used for identity. You can generate an Ethereum private key using any Ethereum wallet, or you can use the utility function `StreamrClient.generateEthereumAccount()`, which returns the address and private key of a fresh Ethereum account.
+
+
+### Subscribing to a stream
+```js 
+
+client.subscribe(STREAM_ID, (message) => {
+    // handle for individual messages
+})
+
+```
+### Creating & publishing to a stream
+```js 
+// [GAS REQUIRED]
+const stream = await client.createStream({
+    id: '/foo/bar'
+})
+
+await stream.publish({foo: 'bar'})
+```
 
-## Installation
 
+### Installation
 The client is available on [npm](https://www.npmjs.com/package/streamr-client) and can be installed simply by:
 
 ```
 npm install streamr-client
 ```
 
-## Usage
+### Importing `streamr-client`
+When using Node.js remember to import the library with:
 
-Here are some usage examples. More examples can be found [here](https://github.com/streamr-dev/examples).
+```js
+const { StreamrClient } = require('streamr-client')
+```
 
-In Streamr, Ethereum accounts are used for identity. You can generate an Ethereum private key using any Ethereum wallet, or you can use the utility function `StreamrClient.generateEthereumAccount()`, which returns the address and private key of a fresh Ethereum account.
+For usage in the browser include the latest build, e.g. by including a `<script>` tag pointing at a CDN:
 
-### Creating a StreamrClient instance
+```html
+<!-- for Brubeck package (6x) -->
+<script src="https://unpkg.com/streamr-client@6.0.0-beta.2/streamr-client.web.js"></script>
+```
+
+>  To use with react please see [streamr-client-react](https://github.com/streamr-dev/streamr-client-react)
+
+### StreamId and the StreamDefinition type
+The [StreamDefinition type](https://github.com/streamr-dev/network-monorepo/blob/04bb2df5b246c502de8e13fbb51b9db1ffb4f668/packages/client/src/types.ts#L9) defines what a stream id should be:
+```typescript
+export type StreamDefinition = string
+    { id: string, partition?: number },
+    ...
+}
+```
+
+This means one can define a stream id for our known `address` at `partition: 1` as either:
+```js
+const streamId = `${address}/foo/bar#1'
+```
+Or:
+```js
+const streamId = {
+    id: `${address}/foo/bar`,
+    partition: 1
+}
+```
```

Comment:
As discussed in meeting, let's avoid referring to `StreamDefinition` directly as it is a technicality from the end user's perspective. Instead something like "The public method of the client generally supports the following three ways of defining a stream:"
- `${address}/foo/bar`
- `${address}/foo/bar#4`
- `{ id:  `${address}/foo/bar`, partition: 4 }`

Reply:
We could do this explaining in partitioning section. And all examples before the partitionin sections would just use the plain string as a streamId (therefore defaulting to 0 partition).